### PR TITLE
BMDA: Fix build on OSX

### PR DESCRIPTION
**Pull request by djix123**
## Detailed description

Fixes BMDA build issues on OSX:

(i) Adds certain compiler flags if not OSX (-Wmaybe-uninitialized -Wstringop-overflow -Wunsafe-loop-optimizations)
(ii) Add 'const' to serial_open declaration on OSX

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds for hardware native (`make PROBE_HOST=native`)
* [x] It builds as BMDA (`make PROBE_HOST=hosted`)
* [x] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

**Comment by zyp** [comment]
This looks good to me. Except the conditional logic, these are the exact changes I've been keeping around in my local working tree.

**Comment by djix123** [comment]
Yes, I've been keeping them around locally too. Thought it about time to submit as a patch!

File: src/Makefile [hunk 1]
```diff
@@ -3,17 +3,25 @@ PLATFORM_DIR = platforms/$(PROBE_HOST)
 VPATH += $(PLATFORM_DIR) target
 ENABLE_DEBUG ?=
 
+CC ?= gcc
```
Comment (dragonmux) [inline src/Makefile:6]: CC is already set in the platforms and needs to use the form `$(CROSS_COMPILE)gcc`. We are fine with the assignment being moved into this Makefile, but it must not semantically change in its form for now.

File: src/Makefile [hunk 2]
```diff
@@ -3,17 +3,25 @@ PLATFORM_DIR = platforms/$(PROBE_HOST)
...
+SYS = $(shell $(CC) -dumpmachine)
+
 ifneq ($(V), 1)
 MAKEFLAGS += --no-print-dir
 Q := @
 endif
 
-CFLAGS += -Wall -Wextra -Werror -Wmaybe-uninitialized \
-	-Wreturn-type -Wstringop-overflow -Wunsafe-loop-optimizations \
+CFLAGS += -Wall -Wextra -Werror \
```
Comment (dragonmux) [inline src/Makefile:14]: Please reflow `-Wreturn-type` back onto this line

**Review by dragonmux — CHANGES_REQUESTED** [review]
Thank yo for the contribution! This looks reasonable, though there are a couple of changes we request get made before we can accept this for v1.9.

Additionally, please follow the contribution guidelines for the commit messages - we suggest the `hosted/serial_unix:` change should be a separate commit, and the Makefile changes are either `makefile:` or `misc:` (probably the latter as in fixing the first review item you'll need to modify all the platform Makefile.inc files)

File: src/Makefile (hunk 1 above)
Comment (djix123) [reply src/Makefile]: when checking I realized there was no need to assign CC in the Makefile as it is already set - so I've removed from the proposed patch

File: src/Makefile (hunk 2 above)
Comment (djix123) [reply src/Makefile]: Done

**Comment by djix123** [comment]
I've reworked the pull request into two separate commits. Let me know if I need to change anything else. Thank you for considering this pull request.

**Review by dragonmux — APPROVED** [review]
LGTM, merging

**Comment by djix123** [comment]
Thank you!

**Pull request merged**